src/systemd/sd-id128.h

```diff
@@ -105,6 +105,9 @@ int sd_id128_get_boot_app_specific(sd_id128_t app_id, sd_id128_t *ret);
 #define SD_ID128_MAKE_STR(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p) \
         #a #b #c #d #e #f #g #h #i #j #k #l #m #n #o #p
 
+#define SD_ID128_MAKE_UUID_STR(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p) \
+        #a #b #c #d "-" #e #f "-" #g #h "-" #i #j "-" #k #l #m #n #o #p
```

Comment:
btw, strictly speaking, this is actually wrong. variant 2 uuids are supposed to be processed in weird microsoft native endian I think. we should probably at least document that we treat all uuids as variant 1, i.e. as RFC4122 uuids, regardless of the encoded variant. This is what most code actually does afaics.

Given we already have SD_ID128_UUID_FORMAT_STR we expose the fact that we treat everything as variant 1 already, so i guess adding this change is totally OK, but probably could gain a comment one day